Javascript2-week2/functionsMovieApp.js

```diff
@@ -0,0 +1,337 @@
+import { movies } from "./moviesList.js";
+import { allComments, addCommentToStorage } from './commentsFunctions.js';
+
+/* FILTERS */ // (UPDATED JS2 WK2) - Added filters section
+
+// Filters movies by genre
+function filterMoviesByGenre(genre) {
+    const filteredMovies = movies.filter(movie => movie.genres.includes(genre));
+    displayMoviesGrid(filteredMovies);
+}; 
+    // Genre filters
+function setupGenreFilters() {
+    const genreFiltersContainer = document.getElementById('genreFilters');
+    const allGenres = [...new Set(movies.flatMap(movie => movie.genres))]; // Get unique genres
+
+    // Sorts genres alphabetically
+    allGenres.sort((a, b) => a.localeCompare(b));
+
+
+    allGenres.forEach(genre => {
+        const button = document.createElement('button');
+        button.innerText = genre;
+        button.onclick = () => filterMoviesByGenre(genre);
+        genreFiltersContainer.appendChild(button);
+    });
+};
+
+// Displays trending movies
+function displayTrendingMovies() {
+    const trendingMovies = movies.filter(movie => movie.trending);
+    displayMoviesGrid(trendingMovies);
+};
+
+// Displays all the movies
+function displayAllMovies() {
+    displayMoviesGrid(movies);
+};
+
+// Event listeners for the buttons
+document.getElementById('allMovies').addEventListener('click', displayAllMovies);
+document.getElementById('trendingMovies').addEventListener('click', displayTrendingMovies);
+
+// Initialize genre filters
+setupGenreFilters();
+
+/* DROPDOWN MENU*/
+function setupDropdown() {
+    let dropdownButton = document.querySelector(".dropdownContent");
+    let dropdownMenu = document.querySelector(".dropdownMenu");
+
+    dropdownButton.addEventListener("click", function () {
+        if (dropdownMenu.style.display === "block") {
+            dropdownMenu.style.display = "none"; // Hide the menu
+        } else {
+            dropdownMenu.style.display = "block"; // Show the menu
+        };
+    });
+
+    // If the user clicks anywhere outside the dropdown, close the dropdown 
+    document.addEventListener("click", function (event) {
+        if (!dropdownButton.contains(event.target) && !dropdownMenu.contains(event.target)) {
+            dropdownMenu.style.display = "none";
+        }
+    });
+};
+
+setupDropdown();
+
+/* SEARCH BAR */ // (UPDATED JS2 WK2) - Added search bar functionality
+function searchMovies(searchText) {
+    return movies.filter(movie => 
+        movie.title.toLowerCase().includes(searchText.toLowerCase())
+    );
+};
+
+document.querySelector(".searchText").addEventListener("input", (event) => {
```

Comment:
Instead of using a separate function to return filtered movies in line 70 and then calling displayMoviesGrid, you can simplify the function. 
```

document.querySelector(".searchText").addEventListener("input", (event) => {
    displayMoviesGrid(movies.filter(movie =>
        movie.title.toLowerCase().includes(event.target.value.trim().toLowerCase())
    ));
});
```